pilot/pkg/keycertbundle/watcher.go

```diff
@@ -67,7 +84,9 @@ func (w *Watcher) SetAndNotify(key, cert, caBundle []byte) {
 	}
 	w.initDone.Store(true)
 	for _, ch := range w.watchers {
-		ch <- w.bundle
+		go func() {
+			ch <- w.bundle
```

Comment:
This seems risky; if we have a bug where a channel is never read we may leak goroutine forever. Probably low risk though.

One idea to make it safe would be to use `ch` as just a `struct{}` carrier to signal to the consumer to then call `w.FetchBundle()` (or similar). Then we can make it have a buffered size of 1 and skip if its full.

Reply:
Good suggestion, just use the channel to notify signal